vertx-auth-mtls/src/main/java/io/vertx/ext/auth/mtls/impl/SpiffeIdentityExtractor.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.vertx.ext.auth.mtls.impl;
+
+import io.vertx.ext.auth.authentication.CredentialValidationException;
+import io.vertx.ext.auth.mtls.CertificateIdentityExtractor;
+
+import java.security.cert.Certificate;
+import java.security.cert.CertificateParsingException;
+import java.security.cert.X509Certificate;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+
+/**
+ * {@link CertificateIdentityExtractor} implementation for SPIFFE certificates for extracting SPIFFE identity.
+ * SPIFFE is a URI, present as part of SAN of client certificates, it uniquely identifies client.
+ */
+public class SpiffeIdentityExtractor implements CertificateIdentityExtractor
+{
+    private static final int SUBJECT_ALT_NAME_URI_TYPE = 6;
+    private static final String SPIFFE_PREFIX = "spiffe://";
+
+    @Override
+    public String identity(Certificate[] certificateChain) throws CredentialValidationException, CertificateParsingException
+    {
+        X509Certificate[] castedCerts = castCertsToX509(certificateChain);
+        if (castedCerts.length == 0)
+        {
+            throw new CredentialValidationException("No based X509Certificate found for validating");
+        }
+        X509Certificate privateCert = castedCerts[0];
```

Comment:
why checking only the first cert from the array?

Reply:
We assume the first cert is always the private key entry in cert chain arrary. I initially had a loop to go through certs and check their type to get private key specific details. But  there is no method to check if an entry is private key or CA cert entry.

Reply:
Let's think more about it. If we go with this route, definitely need to call out the implicit in javadoc.